src/Symfony/Component/Cache/Traits/ChainTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Traits;
+
+use Psr\SimpleCache\CacheInterface;
+use Symfony\Component\Cache\PruneableInterface;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ *
+ * @internal
+ */
+trait ChainTrait
+{
+    /**
+     * @param PruneableInterface[]|CacheInterface[] $pools
+     *
+     * @return bool
+     */
+    public function doPrune(array $pools)
```

Comment:
Not positive this is necessary. If I should just fold the logic into `ChainAdapter` and `ChainCache` it doesn't cause too much duplication; what do others think?